Sources/NIOHTTP1/HTTPTypes.swift

```diff
@@ -225,7 +441,10 @@ public struct HTTPHeaders: CustomStringConvertible {
     ///
     /// - Parameter name: The name of the header field to remove from the block.
     public mutating func remove(name: String) {
-        self.storage[name.lowercased()] = nil
+        if self.storage.remove(name: name) {
+            // We removed something fomr the internal storage and so it is not continous anymore.
+            self.continous = false
```

Comment:
This seems like something the internal storage would know about, not the top-level.

In fact, I think it's strictly computable: the storage is contiguous if each element's start index is two bytes after the previous element's end index. That lets you just count it, though naturally you can also spend a word and keep a boolean that keeps track of this.